language/move-lang/src/parser/syntax.rs

```diff
@@ -1678,6 +1679,162 @@ fn parse_spec_variable<'input>(tokens: &mut Lexer<'input>) -> Result<SpecBlockMe
     ))
 }
 
+// Parse a specification schema include.
+//    SpecInclude = "include" <ModuleAccess> <OptionalTypeArgs> ";"
+fn parse_spec_include<'input>(tokens: &mut Lexer<'input>) -> Result<SpecBlockMember, Error> {
+    let start_loc = tokens.start_loc();
+    consume_name_value(tokens, "include")?;
+    let name = parse_module_access(tokens, || "a schema name".to_string())?;
+    let type_arguments = parse_optional_type_args(tokens)?;
+    consume_token(tokens, Tok::Semicolon)?;
+    Ok(spanned(
+        tokens.file_name(),
+        start_loc,
+        tokens.previous_end_loc(),
+        SpecBlockMember_::Include {
+            name,
+            type_arguments,
+        },
+    ))
+}
+
+// Parse a specification schema apply.
+//    SpecInclude = "apply" <ModuleAccess> <OptionalTypeArgs> "to" Comma<FunctionPattern> ";"
+fn parse_spec_apply<'input>(tokens: &mut Lexer<'input>) -> Result<SpecBlockMember, Error> {
+    let start_loc = tokens.start_loc();
+    consume_name_value(tokens, "apply")?;
+    let name = parse_module_access(tokens, || "a schema name".to_string())?;
+    let type_arguments = parse_optional_type_args(tokens)?;
+    // Check for 'to' Name value token but don't consume it because we need a start token for
```

Comment:
You can use parse_comma_list_after_start so you can go ahead and consume the "to" here.

But, the parse_comma_list functions allow trailing commas. That would be a little strange here, with the semicolon terminator. Do we care about that?

Reply:
Done. I think we can live with users being allowed to write `a,b,;` if they really wish to do such ugly things. It seems consistent with comma usage in the rest of the language.

Reply:
Update: I actually changed this by introducing a new `parse_list` which I also needed for other reasons (arbitrary separator). Now trailing comma is not longer allowed.